app/Http/Controllers/DeviceController.php

```diff
@@ -55,18 +55,20 @@ class DeviceController extends Controller
         'capture' => \App\Http\Controllers\Device\Tabs\CaptureController::class,
     ];
 
-    public function index(Request $request, $device_id, $current_tab = 'overview', $vars = '')
+    public function index(Request $request, $device, $current_tab = 'overview', $vars = '')
     {
-        $device_id = (int)str_replace('device=', '', $device_id);
-        $current_tab = str_replace('tab=', '', $current_tab);
-        $current_tab = array_key_exists($current_tab, $this->tabs) ? $current_tab : 'overview';
+        $device = str_replace('device=', '', $device);
+        $device = is_numeric($device) ? DeviceCache::get($device) : DeviceCache::getByHostname($device);
+        $device_id = $device->device_id;
         DeviceCache::setPrimary($device_id);
-        $device = DeviceCache::getPrimary();
 
         if (!$device->exists) {
             abort(404);
         }
 
+        $current_tab = str_replace('tab=', '', $current_tab);
+        $current_tab = array_key_exists($current_tab, $this->tabs) ? $current_tab : 'overview';
```

Comment:
How about returning 404 instead? Personally I'm no fan of kidnappning invalid urls

Reply:
jumping back to overview, or showing a 404. Both is ok, but it should be globaly equal

Reply:
@Jellyfrog what do you mean?

Reply:
If I go to /foobar it will return overview. It's a catch all, I prefer 404

Reply:
You mean supporting the legacy device URLs?
1. There is not change to that in this PR, so kinda unrelated.
2. Parts of our own code still call the legacy URLs.
3. I'm 100% certain there are many alert templates linking using the legacy URLs.
4. There is a high likely hood external integration uses the legacy url format.

Reply:
I mean this line i commented. If it doesn't find a match in $this->tabs, it returns overview. It might miss something regarding the legacy urls

Reply:
That is the legacy behavior.  Sorry, I was unsure, because as I said, I didn't change these two lines in the PR, just moved them.

Reply:
I would be fine with sending 404, send a different PR for that though.

Reply:
I'll put it on my todo